pyro/distributions/transforms/generalized_channel_permute.py

```diff
@@ -2,38 +2,131 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import torch
-import torch.nn.functional as F
-from torch.distributions import constraints
+from torch.distributions import Transform, constraints
 
+from pyro.distributions.conditional import ConditionalTransformModule
 from pyro.distributions.torch_transform import TransformModule
 from pyro.distributions.util import copy_docs_from
+from pyro.nn import DenseNN
 
 
-@copy_docs_from(TransformModule)
-class GeneralizedChannelPermute(TransformModule):
-    """
+@copy_docs_from(Transform)
+class ConditionedGeneralizedChannelPermute(Transform):
+    domain = constraints.real
```

Comment:
will this be missing needed docstrings?

Reply:
I've been omitting docstrings from classes starting with `Conditioned` because they're helpers that aren't exported. The other two classes that are exported have docstrings though